openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py

```diff
@@ -1,22 +1,161 @@
 
 
 class PathstreamExternalEnrollment(BaseExternalEnrollment):
     """
     PathstreamExternalEnrollment class.
     """
+    def __init__(self):
+        self.client = None
+        self.S3_BUCKET = settings.OEE_PATHSTREAM_S3_BUCKET
+        self.S3_FILE = settings.OEE_PATHSTREAM_S3_FILE
+
     def __str__(self):
         return 'pathstream'
 
+    def _init_s3(self):
+        """Start S3 connection.
+        """
+        self.client = boto3.client(
+            's3',
+            aws_access_key_id=settings.OOE_PATHSTREAM_S3_ACCESS_KEY,
+            aws_secret_access_key=settings.OOE_PATHSTREAM_S3_SECRET_KEY,
+        )
+
+    def _download_file(self):
+        """Download the file from an S3 Bucket.
+
+        :return: True if file was downloaded, else False"""
+        if self.client is None:
+            raise S3NotInitialized('_init_s3 has not been called yet.')
+
+        try:
+            self.client.download_file(
+                self.S3_BUCKET,
+                self.S3_FILE,
+                self.S3_FILE,
+            )
+        except ClientError as error:
+            LOG.error('Failed to download the file to S3. Reason: %s', str(error))
+            raise ExecutionError
+
+        LOG.info('File [%s] was downloaded from S3 for [%s]', self.S3_FILE, self.__str__())
+
+    def _update_file(self, enrollments_data):
+        """Append enrollment data to the downloaded S3 file
+
+        :param enrollments_data: List of the enrollments data
+        :return: True if file was updated, else False"""
+        if os.path.exists(self.S3_FILE):
+            with open(self.S3_FILE, 'a') as f:
+                f.writelines(enrollments_data)
+        else:
+            LOG.error('File does not exist or _download_file has not been called')
+            raise ExecutionError
+
+    def _upload_file(self):
+        """Upload the file to an S3 bucket
+
+        :return: True if file was uploaded, else False
+        """
+        if self.client is None:
+            raise S3NotInitialized('_init_s3 has not been called yet.')
+
+        try:
+            self.client.upload_file(
+                self.S3_FILE,
+                self.S3_BUCKET,
+                self.S3_FILE,
+            )
+        except ClientError as error:
+            LOG.error('Failed to upload the file to S3. Reason: %s', str(error))
+            raise ExecutionError
+
+        LOG.info('File [%s] was uploaded to S3 for [%s]', self.S3_FILE, self.__str__())
+
+    def _delete_downloaded_file(self):
+        """Delete the downloaded file.
+        """
+        if os.path.exists(self.S3_FILE):
+            os.remove(self.S3_FILE)
+        else:
+            LOG.error('File could not be deleted, File does not exist')
+            raise ExecutionError
+
+    def _is_event_able_to_upload(self, event):
+        """This method returns True is the event 'is_uploaded' value is False."""
+        if event.get('is_uploaded') is False:
```

Comment:
@Jacatove, what do you think about this?
```suggestion
        if not event.get('is_uploaded'):
```

Reply:
I think that is closer to the natural language. @ivanvgh